lib/caching/CachedMccClient.ts

```diff
@@ -77,55 +82,60 @@ export class CachedMccClient<T, B> {
     this.retriesPs = retriesPs;
   }
 
-  // returns T or null
-  public getTransactionFromCache(txId: string): Promise<T> | undefined {
+  // Returns transaction from cache
+  public getTransactionFromCache(txId: string): Promise<ITransaction> | undefined {
     return this.transactionCache.get(txId);
   }
 
   public async getTransaction(txId: string) {
-    try {
-      // if(!this.canAccept) {
-      //   throw Error("Cannot accept transaction requests");
-      // }
-      let txPromise = this.getTransactionFromCache(txId);
-      if (txPromise) {
-        return txPromise;
-      }
-      let newPromise = this.client.getTransaction(txId);
-      this.transactionCache.set(txId, newPromise as Promise<any>); // TODO type check
-      this.checkAndCleanup();
-      return newPromise;
-    } catch (error) {
-      logException(error, `CachedMccClient::getTransaction(${txId})`);
+    let txPromise = this.getTransactionFromCache(txId);
```

Comment:
These two functions `getTransactionFromCache` and `getTransaction` are almost identical and should be merged.
into `async getTransaction`.
It makes no sense to add FromCache on end names when the class name is Cached.